### PR TITLE
Refactor/modal

### PR DESCRIPTION
**Pull request by MartinWeb**
## Related issue
[Bug](https://github.com/AxaFrance/react-toolkit/issues/1047)

### Person(s) for reviewing proposed changes

@arnaudforaison @samuel-gomez @JLou @buddyvegas 

# Important

**This pull request introduces breaking changes :** 

- the isOpen property has been removed to match the way of using the dialog element. We do not change visibility of the element anymore, we directly call the showModal or close functions to open and close the dialog
- related to that, i didn't find a way to continue the export of Modal.Header, Modal.Body, etc.... because of the usage of forwardRef. Instead i directly use Header, Body, etc... If someone has a way to make it work i am open to suggestion :)

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import '../packages/all/dist/style/af-toolkit-core.css';
 import '../packages/core/dist/assets/fonts/icons/af-icons.css';
 import './storybook.css';
 
-export const decorators = [addReadme];
+// export const decorators = [addReadme];
```
Comment (GuillaumeKESTEMAN) [inline .storybook/preview.js:8]: Why comment an export and not delete it if it's no longer used ?

File: packages/Modal/boolean/README.md [hunk 2]
```diff
@@ -19,24 +19,35 @@ import '@axa-fr/react-toolkit-button/dist/af-button.css';
...
-      adipisicing ea dolore aute nisi quis ullamco aliquip occaecat. Aute ut
-      mollit amet.
-    </p>
-  </BooleanModal>
-);
-export default BooleanModalDefault;
+import { useRef } from 'react';
+
+export const YourComponent = () => {
+  const ref = useRef < HTMLDialogElement > null;
```
Comment (GuillaumeKESTEMAN) [inline packages/Modal/boolean/README.md:25]:
```suggestion
  const ref = useRef<HTMLDialogElement>(null);
```

File: packages/Modal/default/README.md [hunk 3]
```diff
@@ -24,84 +24,47 @@ import '@axa-fr/react-toolkit-button/dist/af-button.css';
...
-```javascript
-import Modal from '@axa-fr/react-toolkit-modal-default';
-import '@axa-fr/react-toolkit-modal-default/dist/af-modal.css';
-import '@axa-fr/react-toolkit-button/dist/af-button.css';
-```
+import { useRef } from 'react';
 
-### Use
+export const YourComponent = () => {
+  const ref = useRef < HTMLDialogElement > null;
```
Comment (GuillaumeKESTEMAN) [inline packages/Modal/default/README.md:30]:
```suggestion
  const ref = useRef<HTMLDialogElement>(null);
```

File: .storybook/preview.js (hunk 1 above)
Comment (MartinWeb) [reply .storybook/preview.js]: fixed

**Review by samuel-gomez — APPROVED** [review]

**Pull request merged**